components/tag/basic/src/index.js

```diff
@@ -0,0 +1,114 @@
+import React, {Component} from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+
+class TagBasic extends Component {
+  static MAX_LABEL_LENGTH = 100
+
+  /**
+   * @param  {string} options.className custom classname
+   * @param {fn} options.onClick
+   * @return {string} all classnames joined by whitespace
+   */
+  _classNames ({className, onClick}) {
+    return cx(
+      'sui-TagBasic',
+      onClick && 'sui-TagBasic-actionable',
+      className
+    )
+  }
+
+  /**
+   * Cuts off exceeded char limit
+   * @param  {string} label
+   * @return {string}
+   */
+  _truncate (label) {
+    return label.length < TagBasic.MAX_LABEL_LENGTH
+      ? label
+      : label.substr(0, TagBasic.MAX_LABEL_LENGTH)
+  }
+
+  /**
+   * @param {Object} event
+   */
+  _stopEvent (event) {
+    event.preventDefault()
+    event.stopPropagation()
+  }
+
+  /**
+   * @param  {Object} event
+   */
+  onClose = (event) => {
+    this._stopEvent()
+    this.props.onClose()
+  }
+
+  /**
+   * @param  {Object} event
+   */
+  onClick = (event) => {
+    const onClick = this.props.onClick
+    if (!onClick) return
+
+    this._stopEvent(event)
+    onClick()
+  }
+
+  render () {
+    const {
+      className,
+      Icon,
+      onClose,
+      CloseIcon,
+      onClick
+    } = this.props
+
+    const label = this._truncate(this.props.label)
+
+    return (
+      <div
+        className={this._classNames({className, onClick})}
+        onClick={this.onClick}>
+        {
+          Icon &&
+            <span>
+              <Icon svgClass='sui-TagBasic-icon' />
+            </span>
+        }
+        <span className='sui-TagBasic-label' title={label}>
+          {label}
+        </span>
+        {
+          !onClick && onClose &&
```

Comment:
We don't do this anywhere... but it's a good idea to add a warning that's removed on prod build, like react does, when the user pass an `onClick` and `onClose` ? since the second one will be ignored?

Reply:
I guess you are talking about a global solution and not about implementing it in this component, right?

Reply:
Yep. My question was more for @davidbarna & co.

This component could benefit on it, but the implementation should come from outside.

I think of it as QOL since a developer could find himself in a WTF moment when the onclose is ignored etc.

Reply:
Yes, would be nice. For now it's explained in the docs and in the component's own definition. A clickable tag, cannot be closed.